mailbbgun/views.py

```diff
@@ -0,0 +1,160 @@
+import flask
+import jsonschema
+import logging
+import pika
+
+from mailbbgun import app
+from mailbbgun import models
+
+
+if app.config['DEBUG']:
+    logging.basicConfig(level=logging.DEBUG)
+
+
+_DELIVERY_MODE_PERSISTENT = 2
+_LOG = logging.getLogger(__name__)
+
+
+@app.route('/messages', methods=['POST'])
+def new_message():
+    data = _validate_message_request()
+    _LOG.debug("New message: {}".format(data))
+
+    message = models.Message(**data)
+    models.db.session.add(message)
+    models.db.session.commit()
+
+    connection = pika.BlockingConnection(
+        pika.ConnectionParameters(app.config['RABBITMQ_HOST'])
+    )
+    channel = connection.channel()
+    channel.queue_declare(queue='mailbbgun', durable=True)
+    channel.basic_publish(
+        exchange='',
+        routing_key='mailbbgun',
+        body=str(message.id),
+        properties=pika.BasicProperties(
+            delivery_mode=_DELIVERY_MODE_PERSISTENT
+        )
+    )
+    connection.close()
+
+    response = flask.jsonify(message.api_view())
+    response.headers['Location'] = '{}/messages/{}'.format(
+        flask.request.host_url.rstrip('/'), message.id
+    )
+    return response, 201
+
+
+@app.route('/messages', methods=['GET'])
+def list_messages():
+    limit = flask.request.args.get('limit', app.config['DEFAULT_LIMIT'])
+    offset = flask.request.args.get('offset', app.config['DEFAULT_OFFSET'])
+    if int(offset) < 0:
+        flask.abort(400)
```

Comment:
Yes, It would definitely be helpful for clients to receive more specific error messages for 4XX errors, as it would help them debug what is wrong with their request.  I'll noodle on this for a bit and figure out a good patch for it.